tests/Core/RouterTest.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+use PHPUnit\Framework\TestCase;
+use TrendingRepos\Core\Router;
+use TrendingRepos\Core\Request;
+
+class RouterTest extends TestCase
+{
+    protected $router;
+    protected $request;
+
+    public function setUp(): void
+    {
+        $this->request = $this->createMock(Request::class);
+    }
+
+    public function test_catch_exception_when_visiting_wrong_url()
+    {
+        $this->expectExceptionMessage('This is not the web page you are looking for!');
+        $routes = [
+            'GET' => [
+                '/' => 'HomeController@index',
+            ]
+        ];
+        $this->router = new Router($routes, $this->request);
+        $this->get('/wrong-url');
+        $this->router->get();
+    }
+
+    public function test_catch_exception_when_method_in_controller_doesnt_exist()
+    {
+        $this->expectExceptionMessage('This action doesn\'t exist!');
+        $routes = [
+            'GET' => [
+                '/test' => 'HomeController@test',
+            ]
+        ];
+        $this->router = new Router($routes, $this->request);
+        $this->get('/test');
+        $this->router->get();
+    }
+
+    public function test_catch_exception_when_routes_file_is_in_wrong_format()
+    {
+        $this->expectExceptionMessage('Routes file structure in not as we expect');
+        $this->router = new Router(['/' => 'HomeController@home'], $this->request);
+        $this->get('/');
+        $this->router->get();
+    }
+
+    protected function get(string $uri)
+    {
+        $this->call($uri, 'GET');
+    }
+
+    protected function post(string $uri)
```

Comment:
Now it doesn't have to be `protected`, change it to `private`.

Reply:
Think i will get rid of this function for now, We are not using it .

Reply:
Smart choice! 👍